services/src/modules/directives/rest/datasource.spec.ts

```diff
@@ -131,4 +131,39 @@ describe('REST directive data source', () => {
     expect(req.headers.get('Content-Type')).toBe('application/json');
     expect(await req.text()).toBe('{"name":"aviv"}');
   });
+
+  it('Sends request if required query parameters are included', async () => {
```

Comment:
Probably worth adding 1 more case that tests for backward compatibility (the default false value for required) - a query parameter missing the value with the required field not specified at all